### ISSUE TITLE
Console warning on empty nodes

### ISSUE THREAD

**Issue by achimha** [500tech/angular-tree-component#195]
Using version 3.1.0 on Angular 2.3.8 I get a console warning from mobX when supplying an empty array as the `[nodes]` input parameter:

```
[mobx.array] Attempt to read an array index (0) that is out of bounds (0). Please check length first. Out of bound indices will not be tracked by MobX
```

The component should work correctly with an empty input array. A typical use case is provide an empty array and perform an AJAX call to fill it which is then picked up via change detection.

**Comment by dojchek** [comment]
@adamkleingit 
Have the same issue.
Problem seems to be in this [**LOC**](https://github.com/500tech/angular-tree-component/blob/master/lib/components/tree-node-collection.component.ts#L38)

The fix could be as simple as adding one more check for array length:
```
const firstNode = this.viewportNodes && this.viewportNodes.length && this.viewportNodes[0];
```
Not sure if that messes up further functionality, but I guess chances are low since there is a null check in next [**LOC**](https://github.com/500tech/angular-tree-component/blob/master/lib/components/tree-node-collection.component.ts#L39)

**Comment by adamkleingit** [comment]
Hi,
can you make a pull request for this? Thanks

-- 
- Adam Klein <http://il.linkedin.com/in/adamklein500>
- CTO @ 500Tech.com <http://500tech.com/>
- AngularJS, ReactJS, NodeJS
- Development, consulting, training and public speaking
- (+972) 52 747 5633

**Comment by Gillardo** [comment]
let us know if @adamkleingit does PR, need this please

**Comment by dojchek** [comment]
I will do a PR , no problem. Will probably do it today.

**Comment by adamkleingit** [comment]
Merged, will be in the coming version

**Comment by adamkleingit** [comment]
Published in 3.2.4
Thanks
